experimental/dds/tree2/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -141,41 +130,88 @@ class ObjectForest extends SimpleDependee implements IEditableForest {
 				this.forest.activeVisitor = undefined;
 				this.forest.events.emit("afterChange");
 			},
-			onDelete(index: number, count: number): void {
-				this.onMoveOut(index, count);
+			destroy(detachedField: FieldKey, count: number): void {
+				this.forest.invalidateDependents();
+				this.forest.delete(detachedField);
 			},
-			onInsert(index: number, content: Delta.ProtoNode[]): void {
-				const range = this.forest.add(content);
-				moveIn(index, range, this.cursor);
+			create(content: Delta.ProtoNodes, destination: FieldKey): void {
+				this.forest.invalidateDependents();
+				this.forest.add(content, destination);
+			},
+			attach(source: FieldKey, count: number, destination: PlaceIndex): void {
+				this.forest.invalidateDependents();
+				this.attachEdit(source, count, destination);
 			},
-			onMoveOut(index: number, count: number, id?: Delta.MoveId): void {
+			detach(source: Range, destination: FieldKey): void {
 				this.forest.invalidateDependents();
+				this.detachEdit(source, destination);
+			},
+			/**
+			 * Attaches the range into the current field by transferring it from the given source path.
+			 * Does not invalidate dependents.
+			 * @param source - The the range to be attached.
```

Comment:
these comments seem out of date.

Also since count must the the full length, so we need to provide it? Just providing the source detached field should be enough unless we want to use the count for strictly validation purposes (in which case we should document that)

Reply:
I updated the comment with info from the interface's comment. Not sure why we want it explicitly passed but seems like Yann prob had a reason for it so I think we should at least leave it for now.